source/manual/aws-iam-key-rotation.html.md

```diff
@@ -5,93 +5,131 @@ layout: manual_layout
+two keys is when we are in the process of rotating the old one. For this reason,
+AWS allow users to have a maximum of two Access Keys at the same time.
 
-Enter [govuk-secrets/puppet](https://github.com/alphagov/govuk-secrets/tree/master/puppet) in your machine.
-Run one of the [Puppet AWS common actions](https://github.com/alphagov/govuk-secrets/tree/master/puppet_aws#common-actions) and search for the `Access Key ID` in the result.
+If this user has no keys, consider deleting it if it's not needed.
 
-> **Note**
-> The command you should run is related to the environment you need to target.
-> For example, to edit global staging credentials in the `govuk::apps` namespace, you need to run `bundle exec rake 'eyaml:edit[staging,apps]'`.
+If this user has only one key, make a note of the `Access Key ID` and proceed to
+[find out if the key is used].
+
+If this user has two keys, you'll need to follow this guide twice. The first
+time, you'll need to consider one key as the old one to rotate and the other key
+as if it was a new one just created. The second time, you'll follow the guide as
+normal because the user will then have only one key. Now, choose which key is
+the one you are going to rotate and apply the following steps to that key.
+
+### 2. Find out if the key is used
+
+- Check the `Last used` value.
+- If the value is a date, [find where the key is used].
+- If it's `N/A`, you can safely [remove the key] as it was never used.
 
-Now enter [govuk-secrets/puppet_aws](https://github.com/alphagov/govuk-secrets/tree/master/puppet_aws), run the same command again and search for the Access Key ID in the result.
+### 3. Find where the key is used
 
-Keys may also be defined in more then one place, for example the hieradata values
-`govuk::apps::support::aws_access_key_id` and
+- Enter [govuk-secrets/puppet] in your machine.
+- Run the appropriate [Puppet AWS common actions] considering the following:
+  - The command to run is related to the environment you need to target. For
+  example, to edit global staging credentials in the `govuk::apps` namespace,
+  you need to run `bundle exec rake 'eyaml:edit[staging,apps]'`.
+  - You may need to run more than one command. For example, to edit the global
+  integration credentials both in and out of the `govuk::apps` namespace, you
+  need run both `$ bundle exec rake 'eyaml:edit[integration]'` and
+  `$ bundle exec rake 'eyaml:edit[integration,apps]'`.
+- Search for the `Access Key ID` in the result.
+
+Repeat the previous steps for [govuk-secrets/puppet_aws]:
+
+- Enter [govuk-secrets/puppet_aws].
+- Run the same command(s) again.
+- Search for the `Access Key ID` in the result.
+
+Keys may also be defined in more then one place. For example, the hieradata
+values `govuk::apps::support::aws_access_key_id` and
 `govuk::apps::support_api::aws_access_key_id` contain the same key.
+
 Keys may also be shared across AWS and Carrenza environments.
```

Comment:
This line might be redundant as you say above to "Repeat the previous steps for [govuk-secrets/puppet_aws]:".

Reply:
I'd rather make it clear what steps we're talking about.

Reply:
I'm not sure I follow.

But actually, now I'm reading this again, I think I misinterpreted this the first time I read it. If this is trying to say that the key might be present in both the Carrenza and AWS hieradata, it would be probably more useful at the top of this section, as that's something that would be useful to know prior to diving in to the hieradata.

Reply:
Ah, sorry, I thought the comment was related to a different line. I'll move this to the start of this step.

Reply:
I moved it at the top and added an introduction.